File: frontend/lib/models/course.dart
/// Kurs modeli
class Course {
  final int id;
  final String title;
  final String description;
  final int instructorId;
  final String instructorName;
  final double price;
  final String imageUrl;
  final int duration;
  final String level;

  Course({
    required this.id,
    required this.title,
    required this.description,
    required this.instructorId,
    required this.instructorName,
    required this.price,
    required this.imageUrl,
    required this.duration,
    required this.level,
  });

  /// JSON'dan Course nesnesi oluştur
  factory Course.fromJson(Map<String, dynamic> json) {
    return Course(
      id: json['id'],
      title: json['title'],
      description: json['description'],
      instructorId: json['instructorId'],
      instructorName: json['instructorName'],
      price: json['price'].toDouble(),
      imageUrl: json['imageUrl'] ?? '',
      duration: json['duration'] ?? 0,
      level: json['level'] ?? 'Başlangıç',
    );
  }
}

File: backend/src/main/java/com/girisimciturk/miniproje/model/Course.java
package com.girisimciturk.miniproje.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Kurs modeli
 * Sistemdeki eğitim içeriklerini temsil eder
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Course {
    private Long id;
    private String title;              // Kurs başlığı
    private String description;        // Kurs açıklaması
    private Long instructorId;         // Eğitmen ID'si
    private String instructorName;     // Eğitmen adı (kolaylık için)
    private Double price;              // Kurs fiyatı (TL)
    private String imageUrl;           // Kurs görseli URL'i
    private Integer duration;          // Kurs süresi (saat)
    private String level;              // Seviye: Başlangıç, Orta, İleri
}
